### PR TITLE
feat: import orabos

### PR DESCRIPTION
**Pull request by toanju**
* Import orabos features from https://github.com/cobaltcore-dev/gardenlinux to build in the gardenlinux org.
* Adjust build workflow to only build images for `openstackbaremetal-metal3-orabos`, `openstackbaremetal-orabos` and `vmware-orabos` on `amd64`
* Currently GardenLinux `main` is being used

Skipped patches:
* https://github.com/cobaltcore-dev/gardenlinux/commit/63d81112963fec8e5d993c01684848689e0873a6  was merged upstream in https://github.com/gardenlinux/gardenlinux/pull/2386
* https://github.com/cobaltcore-dev/gardenlinux/commit/cda3cb3a4894b62270a04f9a0f01ab81b97c1fec
* https://github.com/cobaltcore-dev/gardenlinux/commit/3f2832744b8d7c75b93c0bc2afc5fbe1af2af723

### CODE REVIEW

File: features/metal3/exec.config [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+cat "$TEMP_DEB_DIR"/build.tar.xz* | xz -d | tar -C "$TEMP_DEB_DIR" -x
+DEBIAN_FRONTEND=noninteractive dpkg -i "$TEMP_DEB_DIR/${package_name}_${tag}_${arch}.deb"
+rm -rf "$TEMP_DEB_DIR"
+
+apt-mark hold ${package_name}
+
+# crictl
+CRI_VERSION=1.27.0
+
```
Comment (gehoern) [inline features/metal3/exec.config:28]: You hardcode the version of the cri. I guess intended. Should be done maybe like the khost feature in Gardenlinux

File: features/metal3/exec.config [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+wget --show-progress --progress=bar:force -4 -P /opt/bin \
+  https://github.com/kubernetes-sigs/cri-tools/releases/download/v${CRI_VERSION}/crictl-v${CRI_VERSION}-linux-amd64.tar.gz
+
```
Comment (gehoern) [inline features/metal3/exec.config:31]: No shasum tests?

File: features/orabos/cloudinit-net/__init__.py [hunk 3]
```diff
@@ -0,0 +1,1409 @@
+# Copyright (C) 2013-2014 Canonical Ltd.
```
Comment (gehoern) [inline features/orabos/cloudinit-net/__init__.py:1]: you could do that via file-include ... then you do not need to copy

File: features/orabos/exec.early [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (gehoern) [inline features/orabos/exec.early:1]: should use glrm (gardenlinux repomanager)

File: features/orabos/exec.post [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (gehoern) [inline features/orabos/exec.post:1]: this copy could be neglected if using file-include

File: features/orabos/file.include/etc/dracut.conf.d/99-disable-watchdog.conf [hunk 6]
```diff
@@ -0,0 +1 @@
+omit_drivers+=" iTCO_wdt "
```
Comment (gehoern) [inline features/orabos/file.include/etc/dracut.conf.d/99-disable-watchdog.conf:1]: what is the issue with the watchdog on your side? does it bevae wrong?

File: features/orabos/file.include/etc/kernel/cmdline.d/10-console.cfg [hunk 7]
```diff
@@ -0,0 +1 @@
+CMDLINE_LINUX="console=tty0 $CMDLINE_LINUX"
```
Comment (gehoern) [inline features/orabos/file.include/etc/kernel/cmdline.d/10-console.cfg:1]: really no serial output ? no remote visibility of errors?

File: features/orabos/file.include/usr/lib/dracut/modules.d/99ensure-hugepages/ensure-hugepages.sh [hunk 8]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
```
Comment (gehoern) [inline features/orabos/file.include/usr/lib/dracut/modules.d/99ensure-hugepages/ensure-hugepages.sh:1]: why you are calculating the hugespage size and do not enforce the "right" size vie kernelcmdline?

File: features/orabos/info.yaml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+description: "ORA Baremetal OS"
```
Comment (gehoern) [inline features/orabos/info.yaml:1]: this combination is difficult. the gardener feature has some bias here that could be avoided by the chost feature.
the other is the combination vhost + chost is that really intended?

File: features/orabos/pkg.exclude [hunk 10]
```diff
@@ -0,0 +1,2 @@
+frr
```
Comment (gehoern) [inline features/orabos/pkg.exclude:1]: does this rally come from us?

Comment (gehoern) [reply features/orabos/pkg.exclude:1]: ah comes from openstackbaremetal feature

**Review by gehoern — APPROVED** [review]

File: features/orabos/exec.post (hunk 5 above)
Comment (NotTheEvilOne) [reply features/orabos/exec.post:1]: See PR https://github.com/gardenlinux/gardenlinux-ccloud/pull/13

File: features/orabos/file.include/usr/lib/dracut/modules.d/99ensure-hugepages/ensure-hugepages.sh (hunk 8 above)
Comment (NotTheEvilOne) [reply features/orabos/file.include/usr/lib/dracut/modules.d/99ensure-hugepages/ensure-hugepages.sh:1]: See PR #12

File: features/metal3/exec.config (hunk 1 above)
Comment (toanju) [reply features/metal3/exec.config:28]: done in https://github.com/gardenlinux/gardenlinux-ccloud/pull/14

File: features/metal3/exec.config (hunk 2 above)
Comment (toanju) [reply features/metal3/exec.config:31]: same

File: features/orabos/file.include/etc/dracut.conf.d/99-disable-watchdog.conf (hunk 6 above)
Comment (toanju) [reply features/orabos/file.include/etc/dracut.conf.d/99-disable-watchdog.conf:1]: watchdog seems to cause issues and did sometimes reset during boot

**Pull request merged**